pkg/sql/logictest/logic.go

```diff
@@ -2966,11 +2973,23 @@ func (t *logicTest) execQuery(query logicQuery) error {
 				// rows in the order in which the query returned them.
 				sortMsg = " -> ignore the following ordering of rows"
 			}
-			fmt.Fprintf(&buf, "but found (query options: %q%s) :\n", query.rawOpts, sortMsg)
-			for _, line := range t.formatValues(actualResultsRaw, query.valsPerLine) {
-				fmt.Fprintf(&buf, "    %s\n", line)
+			var buf bytes.Buffer
+			fmt.Fprintf(&buf, "%s: %s\nexpected:\n%s", query.pos, query.sql, expFormatted.String())
+			fmt.Fprintf(&buf, "but found (query options: %q%s) :\n%s", query.rawOpts, sortMsg, actFormatted.String())
+			if *showDiff {
+				if diff, err := difflib.GetUnifiedDiffString(difflib.UnifiedDiff{
+					A:        difflib.SplitLines(expFormatted.String()),
+					B:        difflib.SplitLines(actFormatted.String()),
```

Comment:
Right now these are based on the raw output, which is a bummer when the column widths change. Perhaps we should use the whitespace normalized version of the rows to produce the diff even though it wouldn't match what we've printed above.